### DISCUSSION TITLE
brew test-bot fails with no such keg error while uninstalling dependencies

### DISCUSSION THREAD

**Discussion by na-trium-144** [Homebrew/discussions#5607] [Tap maintenance and Homebrew development] [answered]
### Output of `brew config`

```shell
N/A
```

### Output of `brew doctor`

```shell
N/A
```

### Description of issue

I'm maintaining a tap at https://github.com/na-trium-144/homebrew-webcface.
Today, I tried to bump version of my formula `webcface@2` as usual and created PR (https://github.com/na-trium-144/homebrew-webcface/pull/102), but CI failed with the following error while uninstalling `crow-unix-socket`, which is a dependency of `webcface@2` and also in the same tap.
I haven't seen such error until previous version bump in last week, and I didn't change anything except bumping version of `webcface@2` (and removing macos-12 from my tests.yml because it seems unsupported now)
I have no idea what to do to fix this. Could anyone help me with this?
```
==> Running Formulae#formula!(na-trium-144/webcface/webcface@2)
==> brew deps --tree --annotate --include-build --include-test na-trium-144/webcface/webcface@2
==> Determining dependencies...
==> brew fetch --formulae --retry aom asciidoc asciidoctor asio autoconf automake bison boost brotli ca-certificates certifi cli11 cmake docbook docbook-xsl doxygen fmt fontconfig freetype gettext ghostscript giflib glib gnu-getopt highway icu4c imath itstool jasper jbig2dec jpeg-turbo jpeg-xl libde265 libgit2 libheif libidn libidn2 liblqr libnghttp2 libomp libpng libraw libssh2 libtiff libtool libunistring libvmaf libxml2 libyaml lit little-cms2 llvm lz4 m4 meson mpdecimal msgpack-cxx na-trium-144/webcface/crow-unix-socket na-trium-144/webcface/curl-ws na-trium-144/webcface/imagemagick-no-openmp ninja openexr openjpeg openssl@3 pcre2 pkg-config python-setuptools python@3.11 python@3.12 readline rtmpdump ruby rust shared-mime-info source-highlight spdlog sphinx-doc sqlite swig utf8cpp webp x265 xmlto xz z3 zstd
==> brew install --only-dependencies --verbose --formula --build-bottle na-trium-144/webcface/webcface@2
==> Starting tests for na-trium-144/webcface/webcface@2
==> brew fetch --formula --retry na-trium-144/webcface/webcface@2 --build-bottle --force
==> brew install --verbose --formula --build-bottle na-trium-144/webcface/webcface@2
==> brew style --formula na-trium-144/webcface/webcface@2
==> brew audit --formula na-trium-144/webcface/webcface@2 --online --git --skip-style
==> brew bottle --verbose --json na-trium-144/webcface/webcface@2 --root-url=https://github.com/na-trium-144/homebrew-webcface/releases/download/webcface@2-2.2.1
==> brew bottle --merge --write --no-commit --no-all-checks ./webcface@2--2.2.1.arm64_sonoma.bottle.json
==> brew uninstall --formula --force na-trium-144/webcface/webcface@2
==> brew uninstall --formulae --force asciidoc asciidoctor asio autoconf automake bison boost certifi cli11 docbook docbook-xsl doxygen gnu-getopt icu4c itstool libgit2 libxml2 libyaml lit llvm meson msgpack-cxx na-trium-144/webcface/crow-unix-socket ninja pkg-config python-setuptools python@3.11 ruby rust source-highlight sphinx-doc swig utf8cpp xmlto z3
==> FAILED
Full uninstall --formulae asciidoc asciidoctor asio autoconf automake bison boost certifi cli11 docbook docbook-xsl doxygen gnu-getopt icu4c itstool libgit2 libxml2 libyaml lit llvm meson msgpack-cxx na-trium-144/webcface/crow-unix-socket ninja pkg-config python-setuptools python@3.11 ruby rust source-highlight sphinx-doc swig utf8cpp xmlto z3 output
  Error: No such keg: /opt/homebrew/Cellar/crow-unix-socket from tap na-trium-144/webcface
```
full log: https://github.com/na-trium-144/homebrew-webcface/actions/runs/10876825484/job/30177300416?pr=102

**Comment by carlocab** [comment]
Thanks for the report. That's a bit weird. It looks like `na-trium-144/webcface/crow-unix-socket` is not installed.

The [exact same step](https://github.com/na-trium-144/homebrew-webcface/actions/runs/10777881991/job/29887982697?pr=101#step:8:42) from your last version bump worked, though.

Seems like it might be due to a change in `brew`. Ideas here, @Homebrew/brew?

**Reply by Bo98** [reply]
`--force` should never raise `No such keg` errors, so it shouldn't matter what is or isn't installed.

I'm however unaware of any changes regarding this.

**Reply by Bo98** [reply]
Ok tracked it down to https://github.com/Homebrew/brew/commit/8cb62b13987087469f2dbdbe622dac01f9ea095c I think. Will fix.

**Comment by carlocab** [comment]
If you checkout your PR locally, and try to run
```
brew install --only-dependencies --verbose --formula --build-bottle na-trium-144/webcface/webcface@2
```
does that install `na-trium-144/webcface/crow-unix-socket`?

**Reply by na-trium-144** [reply]
Yes, it installed crow-unix-socket and I could also uninstall it without error.

**Comment by carlocab** [comment, marked as answer]
@Bo98 should've fixed this in https://github.com/Homebrew/brew/pull/18334. Please try your workflow again and let us know if that works for you.

**Reply by na-trium-144** [reply]
I re-ran the workflow and it worked. Thank you!
